components/omega/src/ocn/Eos.cpp

```diff
@@ -0,0 +1,262 @@
+          });
+   }
+}
+
+//------------------------------------------------------------------------------
+// Define IO fields and metadata
+void Eos::defineFields() {
+
+   int Err = 0;
+
+   SpecVolFldName          = "SpecVol";
+   SpecVolDisplacedFldName = "SpecVolDisplaced";
+   if (Name != "Default") {
+      SpecVolFldName.append(Name);
+      SpecVolDisplacedFldName.append(Name);
+   }
+
+   // Create fields for state variables
+   int NDims = 2;
+   std::vector<std::string> DimNames(NDims);
+   DimNames[0] = "NCells";
+   DimNames[1] = "NVertLevels";
+
+   auto SpecVolField =
+       Field::create(SpecVolFldName,                   // Field name
+                     "Layer-averaged Specific Volume", /// long Name
+                     "m3 kg-1",                        // units
+                     "sea_water_specific_volume",      // CF-ish Name
+                     0.0,                              // min valid value
+                     9.99E+30,                         // max valid value
+                     -9.99E+30, // scalar used for undefined entries
+                     NDims,     // number of dimensions
+                     DimNames   // dimension names
+       );
+   auto SpecVolDisplacedField =
+       Field::create(SpecVolDisplacedFldName, // Field name
+                     "Specific Volume displaced adiabatically "
+                     "to 1 layer below",                    /// long Name
+                     "m3 kg-1",                             // units
+                     "sea_water_specific_volume_displaced", // CF-ish Name
+                     0.0,                                   // min valid value
+                     9.99E+30,                              // max valid value
+                     -9.99E+30, // scalar used for undefined entried
+                     NDims,     // number of dimensions
+                     DimNames   // dimension names
+       );
+
+   // Create a field group for the eos-specific state fields
+   EosGroupName = "Eos";
+   if (Name != "Default") {
+      EosGroupName.append(Name);
+   }
+   auto EosGroup = FieldGroup::create(EosGroupName);
+
+   // Add restart group if needed
+   if (!FieldGroup::exists("Restart"))
+      auto RestartGroup = FieldGroup::create("Restart");
+
+   Err = EosGroup->addField(SpecVolDisplacedFldName);
+   if (Err != 0)
+      LOG_ERROR("Error adding {} to field group {}", SpecVolDisplacedFldName,
+                EosGroupName);
+   Err = EosGroup->addField(SpecVolFldName);
+   if (Err != 0)
+      LOG_ERROR("Error adding {} to field group {}", SpecVolFldName,
+                EosGroupName);
+
+   Err = FieldGroup::addFieldToGroup(SpecVolDisplacedFldName, "Restart");
+   if (Err != 0)
+      LOG_ERROR("Error adding {} to Restart field group",
+                SpecVolDisplacedFldName);
+   Err = FieldGroup::addFieldToGroup(SpecVolFldName, "Restart");
+   if (Err != 0)
+      LOG_ERROR("Error adding {} to Restart field group", SpecVolFldName);
```

Comment:
I don't know that we need to add this to the restart file... in fact, I don't think this should be a saved variable at all. Its really just something to be used in situ when needed, since the displaced amount can change depending on the application. Thoughts?

Reply:
I'd have to look closer at the equations, but do we need this variable in calculations prior to calling the EOS?  If so we may need SpecVol on restart.

Reply:
Oh sorry @vanroekel, I meant just the displaced value, so `SpecVolDisp`. I assume we will want `SpecVol`.

Reply:
If the state variables are bfb in restarts and the EOS is reproducible, I don't think any of the EOS fields need to be in restart - they can be re-calculated after the state has been read.

Reply:
Done!